taskvine/src/worker/vine_cache.c

```diff
@@ -711,8 +699,14 @@ vine_cache_status_t vine_cache_ensure(struct vine_cache *c, const char *cachenam
 			LIST_ITERATE(f->mini_task->input_mounts, m)
 			{
 				result = vine_cache_ensure(c, m->file->cached_name);
-				if (result != VINE_CACHE_STATUS_READY)
+				if (result == VINE_CACHE_STATUS_FAILED) {
```

Comment:
This may not come into play right away, but based on our prior discussion, it seems that `FAILED` may quickly transition to `UNKNOWN` when cleaned up, and so they should be treated as equivalent.